### PR TITLE
Allow the AWS_DEFAULT_REGION environment variable

### PR DESCRIPTION
**Pull request by ScruffyProdigy**
Amazon has this variable documented, and apparently people are trying to use it, so we should support it

@nicolasnoble

### CODE REVIEW

File: src/core/lib/security/credentials/external/aws_external_account_credentials.cc [hunk 1]
```diff
@@ -140,6 +141,10 @@ void AwsExternalAccountCredentials::RetrieveSubjectToken(
 
 void AwsExternalAccountCredentials::RetrieveRegion() {
   UniquePtr<char> region_from_env(gpr_getenv(kRegionEnvVar));
+  if (region_from_env == nullptr) {
+    UniquePtr<char> default_region_from_env(gpr_getenv(kDefaultRegionEnvVar));
```
Comment (markdroth) [inline src/core/lib/security/credentials/external/aws_external_account_credentials.cc:145]: This variable looks unnecessary.  Instead, just say `region_from_env = MakeUnique(gpr_getenv(kDefaultRegionEnvVar))`.

File: src/core/lib/security/credentials/external/aws_external_account_credentials.cc [hunk 2]
```diff
@@ -30,6 +30,7 @@ namespace {
 const char* kExpectedEnvironmentId = "aws1";
 
 const char* kRegionEnvVar = "AWS_REGION";
+const char* kDefaultRegionEnvVar = "AWS_REGION";
```
Comment (markdroth) [inline src/core/lib/security/credentials/external/aws_external_account_credentials.cc:33]: Shouldn't this say `AWS_DEFAULT_REGION`?

**Review by markdroth — CHANGES_REQUESTED** [review]

File: src/core/lib/security/credentials/external/aws_external_account_credentials.cc (hunk 1 above)
Comment (ScruffyProdigy) [reply src/core/lib/security/credentials/external/aws_external_account_credentials.cc]: It looks like MakeUnique was removed in favor of absl::make_unique.  absl::make_unique didn't seem to work with any of <char>, <char *>, <char []>, but just using UniquePtr<char>() seems to work here

File: src/core/lib/security/credentials/external/aws_external_account_credentials.cc (hunk 2 above)
Comment (ScruffyProdigy) [reply src/core/lib/security/credentials/external/aws_external_account_credentials.cc]: yup

**Comment by markdroth** [comment]
This looks good.  I've triggered the tests, so let's see how they look.

**Comment by markdroth** [comment]
It looks like there are a bunch of test failures that you'll need to fix before we can merge this.

**Review by markdroth — APPROVED** [review]

**Review by vjpai — COMMENTED** [review]
LGTM post-merge

**Pull request merged**